File: Frontend/src/models/group.ts
import { Role } from 'models/role';


export interface Group {
    id: number,
    description: string,
    active: boolean
    role: Role,
    startDate: Date,
    endDate: Date
}

/*
export class Group{

    constructor(
        public role: Role,
        public id: number = 0,
        public description: string = "",
        public active: boolean = false,
        public startDate: Date = null,
        public endDate: Date = null
    ){}
}
*/

File: Backend/src/Sofco.Model/Models/Admin/Group.cs
using System;
using System.Collections.Generic;
using Sofco.Model.Interfaces;
using Sofco.Model.Relationships;

namespace Sofco.Model.Models.Admin
{
    public class Group : BaseEntity, ILogicalDelete, IAuditDates
    {
        public string Description { get; set; }

        public bool Active { get; set; }

        public Role Role { get; set; }

        public IList<UserGroup> UserGroups { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime? EndDate { get; set; }

        public void ApplyTo(Group item)
        {
            item.Id = this.Id;
            item.Description = this.Description;
            item.Active = this.Active;
        }
    }
}
